lib/languages/js/source-code/source-code.js

```diff
@@ -264,103 +264,37 @@ function findLineNumberBinarySearch(lineStartIndices, target) {
  * Ensures that variables representing built-in properties of the Global Object,
  * and any globals declared by special block comments, are present in the global
  * scope.
- * @param {Scope} globalScope The global scope.
+ * @param {ScopeManager} scopeManager Scope manager.
  * @param {Object|undefined} configGlobals The globals declared in configuration
  * @param {Object|undefined} inlineGlobals The globals declared in the source code
  * @returns {void}
  */
 function addDeclaredGlobals(
-	globalScope,
-	configGlobals = {},
-	inlineGlobals = {},
+	scopeManager,
+	configGlobals = Object.create(null),
+	inlineGlobals = Object.create(null),
 ) {
-	// Define configured global variables.
-	for (const id of new Set([
-		...Object.keys(configGlobals),
-		...Object.keys(inlineGlobals),
-	])) {
-		/*
-		 * `normalizeConfigGlobal` will throw an error if a configured global value is invalid. However, these errors would
-		 * typically be caught when validating a config anyway (validity for inline global comments is checked separately).
-		 */
-		const configValue =
-			configGlobals[id] === void 0
-				? void 0
-				: normalizeConfigGlobal(configGlobals[id]);
-		const commentValue = inlineGlobals[id] && inlineGlobals[id].value;
-		const value = commentValue || configValue;
-		const sourceComments = inlineGlobals[id] && inlineGlobals[id].comments;
-
-		if (value === "off") {
-			continue;
-		}
-
-		let variable = globalScope.set.get(id);
-
-		if (!variable) {
-			variable = new eslintScope.Variable(id, globalScope);
-
-			globalScope.variables.push(variable);
-			globalScope.set.set(id, variable);
-		}
+	const finalGlobals = Object.assign(Object.create(null), configGlobals);
```

Comment:
Maybe simpler?

```suggestion
	const finalGlobals = { __proto__: null, ...configGlobals };
```

Reply:
It's possible that V8 improved performance of spreading and object creation over time. Interestingly there are even older discussions on the Internet (like [this one](https://www.reddit.com/r/node/comments/rp6jb8/discussion_object_spread_much_faster_than/)) pointing out that object spread is faster. Anyway, this is nothing important in this case, I just noticed the awkward syntax while reviewing.